api/v1/model_handlers.go

```diff
@@ -318,18 +318,27 @@ func (a *Api) containerPaths(request *restful.Request, response *restful.Respons
 // allNodes returns a list of all the available node names in the cluster.
 func (a *Api) allNodes(request *restful.Request, response *restful.Response) {
 	cluster := a.manager.GetCluster()
+	if cluster == nil {
+		response.WriteErrorString(400, "The model is not activated")
```

Comment:
I can't help but think that these three lines are unnecessarily duplicated across the file. What about a new method/function, something like:

```go
func (a *Api) getNonNilCluster() (model.Cluster, error) {
    cluster := a.manager.GetCluster()
    if cluster == nil {
        return nil, fmt.Errorf("the model is not activated")
    }
    return cluster
}
```

And then:

```go
cluster, err := a.getNonNilCluster()
if err != nil {
    response.WriteError(400, err)
}
[...]
```